### DISCUSSION TITLE
Accessing `this` on head method using Options API

### DISCUSSION THREAD

**Discussion by Geometrically** [nuxt/nuxt#18424] [Questions] [answered]
I am in the process of migrating my site from Nuxt 2 to Nuxt 3 for my app. Part of this involved the heavy use of `this` to access component data (ex: something set in the fetch hook, or a prop passed down from a child page).

However, in Nuxt 3, using the head method now causes my application to error since it is not present. I was wondering if there was an alternative to this. Our pages are fairly huge so I would prefer not to migrate to the composition API (atm). Is there any alternative?

Here's an example of what we were doing in Nuxt 2:

```js
export default {
  async asyncData(data) {
    try {
      const [user, projects] = (
        await Promise.all([
          data.$axios.get(`user/${data.params.id}`, data.$defaultHeaders()),
          data.$axios.get(
            `user/${data.params.id}/projects`,
            data.$defaultHeaders()
          ),
        ])
      ).map((it) => it.data)

      return {
        user,
        projects,
      }
    } catch {
      data.error({
        statusCode: 404,
        message: 'User not found',
      })
    }
  },
  head() {
    return {
      title: this.user.username + ' - Modrinth',
      // etc
    }
  },
}
```
  
  Now, in Nuxt 3, this solution no longer works:
  ```js
  export default defineNuxtComponent({
  async asyncData() {
    const data = useNuxtApp()
    const route = useRoute()

    try {
      const [user, projects] = (
        await Promise.all([
          useBaseFetch(`user/${route.params.id}`, data.$defaultHeaders()),
          useBaseFetch(
            `user/${route.params.id}/projects`,
            data.$defaultHeaders()
          ),
        ])
      }

      return {
        user: ref(user),
        projects: ref(projects),
      }
    } catch {
      createError({
        fatal: true,
        statusCode: 404,
        message: 'User not found',
      })
    }
  },
  // this now causes a render error since this is undefined :(
  head() {
    return {
      title: this.user.username + ' - Modrinth',
      // etc
    }
  },
}
```

I understand this is intended behavior, but I was wondering if there was an alternative to this (a way to access component properties and data to set head data using the options API) without rewriting each page using the composition API (which isn't ideal, due to the size of our codebase- it would be a lot more work).

Thank you!

**Comment by Baroshem** [comment, marked as answer]
Hey,

What if you replace `this.user.username` with something like `user.value.username`?

You can also try to use the builtin meta components if you dont want to use the composables approach but I am not sure if that's what you are looking for.

https://nuxt.com/docs/migration/meta#example-built-in-meta-components

**Reply by Geometrically** [reply]
Thanks, that makes sense and both approaches work!
